File: 코딩테스트 대비/1806.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
   int n, m;
   cin >> n >> m;
   vector<int> v(n);
   for (int i = 0; i < n; ++i)
      cin >> v[i];
   int answer = 999999999, l = 0, r = 1, s = v[0];
   while (r < n or s >= m)
   {
      if (s < m)
         s += v[r++];
      else
      {
         answer = answer > r - l ? r - l : answer;
         s -= v[l++];
      }
   }
   cout << (answer == 999999999 ? 0 : answer);

   return 0;
}

File: 코딩테스트 대비/1806.py
n, m = map(int, input().split())
arr = list(map(int, input().split()))
l, r = 0, 0
s = 0
answer = 999999
while r < n:
    if s < m:
        s += arr[r]
        r += 1
    else:
        s -= arr[l]
        l += 1
    if s >= m:
        answer = min(answer , r - l)
while l < n:
    s -= arr[l]
    l += 1
    if s >= m:
        answer = min(answer , r - l)

print(answer if answer != 999999 else 0)